server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java

```diff
@@ -321,13 +330,23 @@ private String getRemoteAddr() {
         }
     }
 
-    private List<MailAddress> getSenderAsList(Mail originalMail) {
-        MaybeSender reversePath = originalMail.getMaybeSender();
+    private List<MailAddress> getSenderAsList(Mail originalMail) throws MessagingException {
+        Optional<MailAddress> returnPath = Optional.ofNullable(originalMail.getMessage().getHeader(RETURN_PATH))
+                .flatMap(path -> path.length > 0 ? Optional.ofNullable(path[0]) : Optional.empty())
+                .map((Throwing.<String, MailAddress>function(MailAddress::new).sneakyThrow()));
 
-        if (getInitParameters().isDebug()) {
-            LOGGER.debug("Processing a bounce request for a message with a reverse path.  The bounce will be sent to {}", reversePath.asString());
+        if(returnPath.isPresent()) {
+            if (getInitParameters().isDebug()) {
+                LOGGER.debug("Processing a bounce request for a message with a reverse path.  The bounce will be sent to {}", returnPath.get().asString());
```

Comment:
reverse path -> ReturnPath

Let's have a consistent vocabulary to not complexify domain knwoledge...